src/python_testing/TC_ACL_2_6.py

```diff
@@ -179,55 +180,38 @@ async def test_TC_ACL_2_6(self):
         )
         logging.info(f"Read events response: {events_response}")
 
-        if events_response:
-            # Get the most recent events from the read response
-            read_events = sorted([e.Data for e in events_response],
-                                 key=lambda x: next(e.Header.EventNumber for e in events_response if e.Data == x),
-                                 reverse=True)[:2]
-
-        # Verify we got the expected number of events
         asserts.assert_true(events_response, "Did not receive a response when calling ReadEvents")
-        asserts.assert_true(len(read_events) >= 2,
-                            f"Expected at least 2 events from read, but got {len(read_events)}")
-
-        # Verify both read and subscription events match our expectations
-        for event_source in [received_subscription_events, read_events]:
-            for event_data in event_source:
-                found = False
-                for acl_entry in acl_entries:
-                    if event_data.latestValue == acl_entry:
-                        found = True
-                        break
-                asserts.assert_true(found, "Event data doesn't match any expected ACL entry")
-
-        # Log events for debugging
-        logging.info("Most recent subscription events:")
-        for event in received_subscription_events:
-            logging.info(f"  {event}")
-
-        logging.info("Most recent read events:")
-        for event in read_events:
-            logging.info(f"  {event}")
-
+        read_events = sorted([e.Data for e in events_response],
+                             key=lambda x: next(e.Header.EventNumber for e in events_response if e.Data == x))
+
+        if force_legacy_encoding:
```

Comment:
I suppose this is a question for @Alami-Amine , but do you know if the ACL cluster is required to emit events after each list operation, or can it bundle them? I know the SDK does, but I wonder if the other implementations might emit events only at the end and if that would be OK.

Reply:
I know it does in the SDK, but I suppose you could also think of the entire IM action as a single change.

Reply:
Let me pull in the resident expert on alternate interpretations :)

Reply:
Just to complete the circle, the resident expert was @Apollon77, sent this PR over on slack.